### PR TITLE
[DOCS] Edit replace processor docs and publish them

### LINKED ISSUE #33757: Document replace processor
**Describe the enhancement:**

The replace processor is documented [here](https://github.com/elastic/beats/blob/main/libbeat/processors/actions/docs/replace.asciidoc), but was never added to the TOC, so the docs were not published. Was this intentional? Do we want to add the processor to the published docs?

**Describe a specific use case for the enhancement or feature:**
Users might want to use this processor, but can't if it's not documented.

### PR DESCRIPTION
**Pull request by dedemorton**
Closes #33757 

Preview link: https://beats_33775.docs-preview.app.elstc.co/guide/en/beats/filebeat/master/replace-fields.html

Summary:
- Adds `replace` processor to the doc build (it was not being built previously)
- Adds some edits.
- Adds some structural changes to match the new format I'm using for processor docs in this PR: https://github.com/elastic/observability-docs/pull/500

Please respond to the questions/comments that being with // REVIEWERS.

### CODE REVIEW

**Comment by dedemorton** [comment]
@faec Would you mind reviewing my changes in this PR when the preview is finished building? I've changed the wording in the docs because the original text was a bit confusing.

File: libbeat/processors/actions/docs/replace.asciidoc [hunk 1]
```diff
@@ -5,25 +5,24 @@
...
 
-The `replace` processor cannot be used to replace value with a completely new value.
+TIP: You can use this processor to truncate a field value or replace
+it with a new string value. You can also use this processor to mask PII
+information.
 
-TIP: The `replacement` field value can be used to truncate the `field` value or replace
-it with a new string. It can also be used for masking PII information.
+[discrete]
+== Example
```
Comment (dedemorton) [inline libbeat/processors/actions/docs/replace.asciidoc:20]:
```suggestion
==== Example
```

File: libbeat/processors/actions/docs/replace.asciidoc [hunk 2]
```diff
@@ -33,16 +32,35 @@ processors:
       fail_on_error: true
 -------
 
-The `replace` processor has following configuration settings:
+[discrete]
+== Configuration settings
```
Comment (dedemorton) [inline libbeat/processors/actions/docs/replace.asciidoc:36]:
```suggestion
==== Configuration settings
```

File: libbeat/processors/actions/docs/replace.asciidoc [hunk 3]
```diff
@@ -5,25 +5,24 @@
...
-`replacement: replacement-string`, where:
+The `replace` processor takes a list of fields to search for a matching
+value and replaces the matching value with a specified string.
 
-* `field` is the original field name. It's supported to use `@metadata.` prefix for the fields and replace values in the event metadata instead of event fields.
-* `pattern` is the regex pattern to match the field's value
-* `replacement` is the replacement string to use to update the field's value
+// Reviewers: The following sentence was in the original doc, but doesn't make sense to me. should I remove it?
+
+//The `replace` processor cannot be used to replace value with a completely new value.
```
Comment (rdner) [inline libbeat/processors/actions/docs/replace.asciidoc:13]: I agree, it does not make sense. I think what it's supposed to be is:

```suggestion
The `replace` processor cannot be used to create a completely new value.
```

The code reference: 
https://github.com/elastic/beats/blob/26f5d5c0d4ce559b1600dcfdbea0316006ace551/libbeat/processors/actions/replace.go#L99-L119

**Review by rdner — APPROVED** [review]

**Pull request merged**